### ISSUE TITLE
Pick-up SONAME merge requests

### ISSUE THREAD

**Issue by noloader** [weidai11/cryptopp#71]
Todd Knarr has offered a patch for SONAME under Linux. The patch embeds the SONAME in the shared object, names the output artifact according to Linux standards, and symlinks to to the long shared object name.

We want to pick this up in about 30 days, distributions have time to uptake 5.6.3.

**Comment by tknarr** [comment]
Merged 'master' into the branch bringing it up-to-date.
Made the change to running ldconfig and the public_service message, added that to the pull request.

**Comment by noloader** [comment]
See commit f84b3964ee62d0ef .

**Comment by noloader** [comment]
@tknarr - A quick question... Around line 425:

```
ifeq ($(HAS_SOLIB_VERSION),1)
	-$(RM) $(DESTDIR)$(PREFIX)/lib/libcryptopp.so$(SOLIB_COMPAT_SUFFIX)
	-$(RM) $(DESTDIR)$(PREFIX)/lib/libcryptopp.so
	$(LDCONF) $(PREFIX)/lib
endif
```

Should the `ldconf` line include `$(DESTDIR)`? If so, would you mind providing a pull request?

Sorry to ask... I honestly don't know.  `$(DESTDIR)` was an informal request from the Macports folks to integrate into their workfows. Debian and Fedora *may* use it, so I'd like it to be as fit as possible for them.

**Comment by noloader** [comment]
@tknarr - Check this out:

```
$ make libcryptopp.a libcryptopp.so
...
make: *** No rule to make target 'libcryptopp.so'.  Stop.
```

Maybe something like:

```
-libcryptopp.so$(SOLIB_VERSION_SUFFIX): public_service | $(LIBOBJS)
+libcryptopp.so libcryptopp.so$(SOLIB_VERSION_SUFFIX): public_service | $(LIBOBJS)
```

**Comment by tknarr** [comment]
Yes, it should include DESTDIR. I'll fix that and also the makefile target problem.
I can do a pull request for you.

**Comment by tknarr** [comment]
Fixes in pull request #90
